packages/search-ui/src/helpers.js

```diff
@@ -23,5 +23,14 @@ export function removeSingleFilterValue(filters, name, value, filterType) {
 }
 
 export function matchFilter(filter1, filter2) {
-  return deepEqual(filter1, filter2);
+  if (
+    filter1 &&
+    filter1.name &&
+    filter2 &&
+    filter2.name &&
+    filter1.name === filter2.name
```

Comment:
Just for my own context-building - what was the impetus behind adding this extra `.name` checking? Is it fixing an existing issue or is it a performance enhancement?